Release notes tool.

williamspatrick opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Review on Reviewable"/>](https://reviewable.io/reviews/open-power/op-build/185)
<!-- Reviewable:end -->

williamspatrick commented:
@scprek - Please review.

scprek commented on openpower/scripts/release-notes:
```diff
@@ -0,0 +1,152 @@
+my $begin_release = shift;
+my $end_release = shift;
+
+foreach my $repo (keys %{$repos})
```
One thought is if you care to have the order the same every release? Does this stay in the same order each time?

stewart-ibm replied:
It'll be okay as this is just for fetching the repos, the bit that outputs everything is around line 75, which sorts the keys.

scprek replied:
Oh whoops, looks like it's sorted on line 75.

scprek commented on openpower/scripts/release-notes:
```diff
@@ -0,0 +1,152 @@
+
+system("cd op-build; git checkout $end_release --force; git reset HEAD --hard");
+
+open(OP_SUMMARY, "cd op-build; git diff $begin_release...$end_release ".
```
Is git diff supposed to have 3 '.' between $begin_release and $end_release, assuming this works since things look good.

stewart-ibm commented on openpower/scripts/release-notes:
```diff
@@ -0,0 +1,152 @@
+    my $package = $repos->{$repo}->{PACKAGE};
+
+    # Display patches.
+    if (open(LSLOG, "ls op-build/$dir/*.patch | ".
```
although this list of patches should likely be sorted, and it'd be better if using readdir like in the perldoc -tf readdir example:
          opendir(my $dh, $some_dir) || die "can't opendir $some_dir: $!";
                @dots = grep { /^\./ && -f "$some_dir/$_" } readdir($dh);
                closedir $dh;

(although that's a bit of a nitpick)

scprek replied:
readdir is in no particular order, In my experience I've  used a http://en.wikipedia.org/wiki/Schwartzian_transform to get the order I want. However also not sure if sorting needed.

scprek commented:
Looks good to me, just some pickier things that may or may not need to be addressed.

stewart-ibm commented:
@williamspatrick what do we want to do with this tool? Merge it as-is and iterate on it as needed?

williamspatrick commented:
@stewart-ibm - I uploaded my latest version.  I think we can merge it for v1.10 now.

williamspatrick commented:
Biggest changes since last review are:

   1. Added additional repositories.
   2. Added support for `$(pkg)_LATEST_VERSION` that was introduced by Skiboot.

stewart-ibm commented:
LGTM. Any further improvements can occur in tree.
